gems/pending/appliance_console.rb

```diff
@@ -161,7 +182,24 @@ module ApplianceConsole
         say("DHCP Network Configuration\n\n")
         if agree("Apply DHCP network configuration? (Y/N): ")
           say("\nApplying DHCP network configuration...")
-          Env['DHCP'] = true
+          # Remove search and nameserver lines from resolv.conf
+          resolv = LinuxAdmin::Dns.new
+          resolv.search_order = []
+          resolv.nameservers = []
+          resolv.save
+
+          # Enable DHCP for the interface
+          eth0.enable_dhcp
+          eth0.save
+
+          # Restart networking service
+          LinuxAdmin::Service.new("network").restart
+
+          # Get the new address and update /etc/hosts
+          eth0.reload
+          dhcp_ip = eth0.address
+          update_hostname_ip(host, dhcp_ip)
```

Comment:
We could move it to LinuxAdmin, but this is more of a hosts operation and I wouldn't want something that similar to the `LinuxAdmin::NetworkInterface#enable_dhcp` method.

So maybe `LinuxAdmin::Hosts.new.update_ip_for_hostname(hostname, ip)`?